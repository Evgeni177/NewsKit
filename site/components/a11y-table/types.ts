import {TableProps} from '../table';

export interface A11yTableProps {
  name: string;
  description: string;
  table: TableProps;
}
