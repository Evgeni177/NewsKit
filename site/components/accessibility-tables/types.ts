export interface AccessibilityTablesProps {
  focusOrder?: {
    name: string;
    description: string;
    table: {
      columns: string[];
      rows: FocusOrderProps[];
    };
  };
  aria?: {
    name: string;
    description: string;
    table: {
      columns: string[];
      rows: AriaProps[];
    };
  };
  interaction?: {
    name: string;
    description: string;
    table: {
      columns: string[];
      rows: InteractionProps[];
    };
  };
}

export interface FocusOrderProps {
  order: string[];
  element: string | JSX.Element;
  role?: string;
}

export interface AriaProps {
  category: string;
  attribute: string;
  value: string;
  description: string;
}

export interface InteractionProps {
  command: string[];
  description: string;
}
