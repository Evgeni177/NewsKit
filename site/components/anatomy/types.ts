import {ImageProps} from '../../../src/image/types';

export interface AnatomyProps {
  media: ImageProps;
  list: Array<string>;
}
