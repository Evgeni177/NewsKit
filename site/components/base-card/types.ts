import {CardProps} from 'newskit';

export interface BaseCardProps extends CardProps {
  title?: string;
  children: React.ReactNode;
}
