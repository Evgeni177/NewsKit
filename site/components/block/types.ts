import React from 'react';
import {
  FlexDirectionProperty,
  JustifyContentProperty,
  AlignItemsProperty,
  DisplayProperty,
  OverflowProperty,
  BorderStyleProperty,
  ColorProperty,
  AlignSelfProperty,
  PositionProperty,
} from 'csstype';
import {ColorKeys, TypePresetKeys, SizingKeys, Responsive} from 'newskit';

export interface BlockProps {
  as?: React.ElementType;
  $color?: Responsive<ColorKeys | ColorProperty>;
  $backgroundColor?: Responsive<ColorKeys | ColorProperty>;
  $font?: Responsive<TypePresetKeys>;
  $border?: Responsive<SizingKeys | string>;
  $borderTop?: Responsive<SizingKeys | string>;
  $borderRight?: Responsive<SizingKeys | string>;
  $borderBottom?: Responsive<SizingKeys | string>;
  $borderLeft?: Responsive<SizingKeys | string>;
  $borderStyle?: Responsive<BorderStyleProperty>;
  $borderWidth?: Responsive<SizingKeys | string>;
  $borderColor?: Responsive<ColorKeys | ColorProperty>;
  $margin?: Responsive<SizingKeys | string>;
  $marginTop?: Responsive<SizingKeys | string>;
  $marginRight?: Responsive<SizingKeys | string>;
  $marginBottom?: Responsive<SizingKeys | string>;
  $marginLeft?: Responsive<SizingKeys | string>;
  $padding?: Responsive<SizingKeys | string>;
  $paddingTop?: Responsive<SizingKeys | string>;
  $paddingRight?: Responsive<SizingKeys | string>;
  $paddingBottom?: Responsive<SizingKeys | string>;
  $paddingLeft?: Responsive<SizingKeys | string>;
  $width?: Responsive<SizingKeys | string>;
  $height?: Responsive<SizingKeys | string>;
  $maxWidth?: Responsive<SizingKeys | string>;
  $maxHeight?: Responsive<SizingKeys | string>;
  $minWidth?: Responsive<SizingKeys | string>;
  $minHeight?: Responsive<SizingKeys | string>;
  $display?: Responsive<DisplayProperty>;
  $flexWrap?: Responsive<boolean>;
  $flexDirection?: Responsive<FlexDirectionProperty>;
  $flexGrow?: Responsive<number>;
  $justifyContent?: Responsive<JustifyContentProperty>;
  $alignItems?: Responsive<AlignItemsProperty>;
  $alignSelf?: Responsive<AlignSelfProperty>;
  $position?: Responsive<PositionProperty>;
  $top?: Responsive<SizingKeys | string>;
  $right?: Responsive<SizingKeys | string>;
  $bottom?: Responsive<SizingKeys | string>;
  $left?: Responsive<SizingKeys | string>;
  $overflow?: OverflowProperty;
}
