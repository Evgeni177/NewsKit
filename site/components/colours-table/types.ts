import {TableRow} from '../table';

export interface ComponentColoursTableProps {
  components: ColoursTableProps[];
}

export interface ColoursTableProps {
  title: string;
  summary: string;
  description?: string;
  tabs: {
    header: string;
    rows: TableRow[];
    description: string;
  }[];
}
