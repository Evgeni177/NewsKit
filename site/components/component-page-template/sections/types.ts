export interface IntroductionText {
  introduction: string;
}
