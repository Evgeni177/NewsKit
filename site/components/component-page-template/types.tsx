import {LayoutProps} from '../layout';
import {
  IntroductionSectionProps,
  InteractiveDemoSectionProps,
  AnatomySectionProps,
  OptionsSectionProps,
  StatesSectionProps,
  BehaviorsSectionProps,
  UsageSectionProps,
  AccessibilitySectionProps,
  SEOSectionProps,
  ComponentAPISectionProps,
  ComplianceSectionProps,
  RelatedComponentsSectionProps,
} from './sections';

export interface ComponentPageTemplateProps extends IntroductionSectionProps {
  layoutProps: LayoutProps;
  componentDefaultsKey: string;
  interactiveDemo?: InteractiveDemoSectionProps;
  anatomy?: AnatomySectionProps;
  options?: OptionsSectionProps;
  states?: StatesSectionProps;
  behaviors?: BehaviorsSectionProps;
  usage?: UsageSectionProps;
  accessibility?: AccessibilitySectionProps;
  seo?: SEOSectionProps;
  componentAPI?: ComponentAPISectionProps;
  compliance?: ComplianceSectionProps;
  related?: RelatedComponentsSectionProps;
}
