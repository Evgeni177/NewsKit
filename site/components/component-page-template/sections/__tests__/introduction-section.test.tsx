import {IntroductionSection} from '..';
import {renderToFragmentWithTheme} from '../../../../utils/test-utils';
import {IntroductionSectionProps} from '../types';

jest.mock(
  'newskit',
  require('../../../../utils/test-utils').mockNewsKitComponents('Grid', 'Cell'),
);

jest.mock('../../../page-introduction/page-introduction');
jest.mock('../../../meta/meta');

describe('IntroductionSection', () => {
  test('renders section as expected', () => {
    const props: IntroductionSectionProps = {
      pageIntroduction: {dummy: 'page intro props'} as any,
      meta: {dummy: 'meta props'} as any,
    };

    const fragment = renderToFragmentWithTheme(IntroductionSection, props);
    expect(fragment).toMatchSnapshot();
  });
});
