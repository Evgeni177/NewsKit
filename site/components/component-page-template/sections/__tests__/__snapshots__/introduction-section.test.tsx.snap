// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IntroductionSection renders section as expected 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin-top: -24px;
  padding-top: 24px;
}

<div
    data-comp="NewsKit Cell"
    data-props="{
  \\"xs\\": 12
}"
    data-testid="Cell"
  >
    <section
      class="emotion-0"
      data-toc-indexed="Introduction"
      id="introduction"
    >
      <div
        data-comp="NewsKit Grid"
        data-props="{
  \\"lgMargin\\": \\"space000\\",
  \\"xsRowGutter\\": \\"space000\\"
}"
        data-testid="Grid"
      >
        <div>
          PageIntroduction Component with props:{
  "dummy": "page intro props"
}
        </div>
        <div
          data-comp="NewsKit Cell"
          data-props="{
  \\"xs\\": 12,
  \\"md\\": 10,
  \\"xl\\": 8,
  \\"mdOffset\\": 1
}"
          data-testid="Cell"
        >
          <div>
            Meta Component with props:{
  "dummy": "meta props"
}
          </div>
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
